### PR TITLE
trajectory converter tool using MDTraj

### PR DESCRIPTION
**Pull request by tsenapathi**

### CODE REVIEW

**Comment by simonbray** [comment]
This looks good, but it would be nice to make it more flexible e.g. allow the user to upload either an xtc or trr trajectory, and maybe also do the conversion in either direction. I can work on this if you like @tsenapathi?

File: tools/MDTraj/trajectory_converter.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	<param argument="trr_input" type="data" format='trr' label="TRR structure file."/>
+    </inputs>
+    <outputs>
+        <data format="dcd" name="dcdout" label="dcd output"/>
+        <data format="pdb" name="pdbout" label="pdb output"/>
+    </outputs>
+    <tests>
+        <test>
+            <param name="gro_input" value="solv_ions.gro" />
+            <param name="trr_input" value="topol.top" />
```
Comment (simonbray) [inline tools/MDTraj/trajectory_converter.xml]: This should be a trr file, not a topology

Comment (tsenapathi) [reply tools/MDTraj/trajectory_converter.xml]: Thanks I will fix this

**Comment by tsenapathi** [comment]
@simonbray I will add xtc format. All the analysis tools require a DCD and PDB. Especially BiO3D only use DCD format. That is why I didn't make this to able to do both directions.

**Comment by chrisbarnettster** [comment]
> This looks good, but it would be nice to make it more flexible e.g. allow the user to upload either an xtc or trr trajectory, and maybe also do the conversion in either direction. I can work on this if you like @tsenapathi?

A good idea @simonbray - even though current analysis tools are dcd focussed - this is important functionality !

File: tools/MDTraj/trajectory_converter.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    <requirements>
+        <requirement type="package" version="1.9.1">mdtraj</requirement>
+        <requirement type="package" version="2018.2">gromacs</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+      ln -s '$gro_input' ./groinp.gro &&
+      ln -s '$trr_input' ./trrinp.trr &&
+      mdconvert -o ./dcdout.dcd ./trrinp.trr &&
+      cp ./dcdout.dcd '$dcdout' &&
+      gmx editconf  -f ./groinp.gro -o ./pdbout.pdb &&
```
Comment (bgruening) [inline tools/MDTraj/trajectory_converter.xml]: you can replace pdbout.pdb here with '$pdbout' and save the copy in the next line

File: tools/MDTraj/trajectory_converter.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      cp ./pdbout.pdb '$pdbout'
+    ]]></command>
+    <inputs>
+        <param argument="gro_input" type="data" format='gro' label="GRO structure file."/>
+	<param argument="trr_input" type="data" format='trr' label="TRR structure file."/>
```
Comment (bgruening) [inline tools/MDTraj/trajectory_converter.xml]: indentation

File: tools/MDTraj/trajectory_converter.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    <command detect_errors="exit_code"><![CDATA[
+      ln -s '$gro_input' ./groinp.gro &&
+      ln -s '$trr_input' ./trrinp.trr &&
+      mdconvert -o ./dcdout.dcd ./trrinp.trr &&
+      cp ./dcdout.dcd '$dcdout' &&
+      gmx editconf  -f ./groinp.gro -o ./pdbout.pdb &&
+      cp ./pdbout.pdb '$pdbout'
+    ]]></command>
+    <inputs>
+        <param argument="gro_input" type="data" format='gro' label="GRO structure file."/>
```
Comment (bgruening) [inline tools/MDTraj/trajectory_converter.xml]: label should not end with `.`

**Comment by simonbray** [comment]
Another thought: we can separate structure conversion and trajectory conversion into two separate tools. This is also more flexible if users don't want to do both simultaneously. Happy to implement this as well if you like. :)

**Comment by simonbray** [comment]
Will work on extending this today.

**Comment by bgruening** [comment]
@tsenapathi see #10

**Pull request closed without merging**